src/client.ts

```diff
@@ -506,6 +524,14 @@ export class StreamChat<
       console.warn(
         'Consecutive calls to connectUser is detected, ideally you should only call this function once in your app.',
       );
+
+      // If last connection attempt failed, we reset user connection with new data. Otherwise we just return the old connection
```

Comment:
@mahboubii I added this change thinking about keeping the behavior you would want and adapting it a little bit in order to keep the support for offline mode.

Users would still need to await this call and it would still fail in case the connection fail. So they can try to reconnect and things should work fine. The difference is that, on offline mode, I can ignore that error and would still have the user, etc in the client state.

Reply:
still without the disconnect users can't really retry the connectUser()